Template improvements

mejo- opened the pull request:
### 📝 Summary

* Resolves: #267
* Resolves: #1195 
* Resolves: #1319 

<!-- Write a summary of your change and some reasoning if needed -->

#### 🖼️ Screencast

| Screencast |
| --- | 
| a |

#### 🖼️ Screenshots

🏚️ Before | 🏡 After
---|---
B | A


### 🚧 TODO

- [ ] Tests
- [ ] Copy full pages (with emojis etc,) not only page files

### 🏁 Checklist

- [ ] Code is properly formatted (`npm run lint` / `npm run stylelint` / `composer run cs:check`)
- [ ] [Sign-off message](https://probot.github.io/apps/dco/) is added to all commits
- [ ] Tests (unit, integration and/or end-to-end) passing and the changes are covered with tests
- [ ] Documentation ([README](https://github.com/nextcloud/collectives/blob/main/README.md) or [documentation](https://github.com/nextcloud/collectives/blob/main/docs/)) has been updated or is not required

mejo- commented:
I changed the title for the blank page item in template selection from "Blank" to "Blank page" in the meantime so it's more explicit.

max-nextcloud commented:
:heart_eyes: Just had a look at the screenshots and the screencast so far and I like them a lot!
Super clean and seems very usable.

max-nextcloud reviewed (commented):
:microscope: minor nitpick:
 `fix(pages): Unset trash pages when switching the page `
 is when switching the collective - not when switching the page, right?

max-nextcloud reviewed (commented):
When renaming a template I would expect any click outside the text input to save the new name rather than leaving the input active.

![grafik](https://github.com/user-attachments/assets/edc7317b-6f5e-4a2b-a9fb-9cc572c1ab84)

max-nextcloud commented on lib/Controller/TemplateController.php:
```diff
@@ -0,0 +1,89 @@
+		}, $this->logger);
+	}
+
+	#[NoAdminRequired]
+	public function rename(int $collectiveId, int $id, string $title): DataResponse {
+		return $this->handleErrorResponse(function () use ($collectiveId, $id, $title): array {
+			$userId = $this->getUserId();
+			$pageInfo = $this->templateService->rename($collectiveId, $id, $title, $userId);
+			return [
+				'data' => $pageInfo
+			];
+		}, $this->logger);
+	}
+
+	#[NoAdminRequired]
+	public function setEmoji(int $collectiveId, int $id, ?string $emoji = null): DataResponse {
+		return $this->handleErrorResponse(function () use ($collectiveId, $id, $emoji): array {
+			$userId = $this->getUserId();
+			$templateInfo = $this->templateService->setEmoji($collectiveId, $id, $emoji, $userId);
+			return [
+				'data' => $templateInfo
+			];
+		}, $this->logger);
+	}
```
Why not use the corresponding page actions for this?

max-nextcloud replied:
We talked this through and decided it may be useful to have separate api endpoints in case we want to for example restrict work on templates to admins later on.

mejo- commented:
> When renaming a template I would expect any click outside the text input to save the new name rather than leaving the input active.

Good catch. Fixed this now along with some other problems I spotted in the frontend implementation. Also rebased on latest main.

mejo- commented:
Also added the repair step for migrating old templates now. To test it, add some `Template.md` pages in your collectives, bump the app version in `appinfo/info.xml` and run `occ upgrade`.

marcoambrosini reviewed (approved):
Very nice @mejo- 
A few comments from design, none of them blocking. Adding checkboxes so you can easily convert to issue should you need to.

## Editing state
- [ ] There should be some clear indication that I'm editing a template when I'm doing so. Could we embed something like this in the viewer title slot? (screenshot comes from anytype):
<img width="458" alt="Screenshot 2025-04-30 at 17 16 14" src="https://github.com/user-attachments/assets/965ba95f-23b0-4b94-8cbf-ce421e3177fc" />

Otherwise we could play with the string in the title but I'm worried that it wouldn't be very clear if we do that.

## New document creation flow
- [ ] Template picker should be added to the suggestion bar and not interrupt the flow of creating a new document

## Templates list
- [ ] Remove margins between list items
- [ ] Remove border radius of list items
- [x] Icons should have no background
Example screenshot with these tweaks: <img width="1054" alt="Screenshot 2025-04-30 at 17 29 26" src="https://github.com/user-attachments/assets/537f46de-e3e9-4652-ad98-e47d14c71897" />

mejo- commented:
> Icons should have no background

The icons are a button to edit the emoji for the template. I wonder whether that's still discoverable with a tertiary button without background.

marcoambrosini commented:
I think it's ok since this can be edited while editing the template. I would be ok with removing this functionality here and simplify the interaction with this list

mejo- commented:
> I think it's ok since this can be edited while editing the template. I would also be ok with removing this functionality here and simplify the interaction with this list

Nope, unfortunately that's not the case. Now that we open the template markdown file in the viewer, there's no way to edit the emoji while editing the page.

mejo- commented:
> > I think it's ok since this can be edited while editing the template. I would also be ok with removing this functionality here and simplify the interaction with this list
> 
> Nope, unfortunately that's not the case. Now that we open the template markdown file in the viewer, there's no way to edit the emoji while editing the page.

We could add an action "Set emoji" in the three-dots menu though to highlight the possibility of editing the emoji.

max-nextcloud reviewed (approved):
Thanks a lot! :heart:
